libsql-server/src/lib.rs

```diff
@@ -788,25 +803,74 @@ where
         }
     }
 
-    fn libsql_wal_configurators(
+    async fn libsql_wal_configurators(
         &self,
         base_config: BaseNamespaceConfig,
         client_config: Option<(Channel, Uri)>,
-        join_set: &mut JoinSet<anyhow::Result<()>>,
+        task_manager: &mut TaskManager,
         migration_scheduler_handle: SchedulerHandle,
         scripted_backup: Option<ScriptBackupManager>,
         wal_path: PathBuf,
-    ) -> anyhow::Result<NamespaceConfigurators> {
+    ) -> anyhow::Result<(NamespaceConfigurators, MakeReplicationSvc)> {
         tracing::info!("using libsql wal");
         let (sender, receiver) = tokio::sync::mpsc::channel(64);
-        let registry = Arc::new(WalRegistry::new(wal_path, NoStorage, sender)?);
+        let storage = if let Some(ref opt) = self.db_config.bottomless_replication {
+            if client_config.is_some() {
+                anyhow::bail!("bottomless cannot be enabled on replicas");
+            }
+
+            let config = aws_config::load_defaults(BehaviorVersion::latest()).await;
+            let http_client = HyperClientBuilder::new().build(self.connector.clone().unwrap());
+            let mut builder = config.into_builder();
+            builder.set_http_client(Some(http_client));
+            builder.set_endpoint_url(opt.aws_endpoint.clone());
+            builder.set_retry_config(RetryConfig::standard().with_max_attempts(10).into());
+            builder.set_region(Region::new(
+                opt.region.clone().expect("expected aws region"),
+            ));
+            let cred = Credentials::new(
+                opt.access_key_id.as_ref().unwrap(),
+                opt.secret_access_key.as_ref().unwrap(),
+                None,
+                None,
+                "Static",
+            );
+            builder.set_credentials_provider(Some(SharedCredentialsProvider::new(cred)));
+            let config = builder.build();
+            let backend = S3Backend::from_sdk_config(
+                config,
+                opt.bucket_name.clone(),
+                opt.db_id.clone().expect("expected db id"),
+            )
+            .await?;
+            let config = AsyncStorageInitConfig {
+                backend: Arc::new(backend),
+                max_in_flight_jobs: 16,
+            };
+            let (storage, storage_loop) = AsyncStorage::new(config).await;
+
+            task_manager.spawn_with_shutdown_notify(|_| async move {
+                storage_loop.run().await;
+                Ok(())
+            });
+
+            Either::A(storage)
```

Comment:
I think we want to configure it this way so that it doesn't double load the config. I need to look deeper if this code actually does what bottomless did before, but in [this](https://github.com/tursodatabase/libsql/blob/main/bottomless/src/replicator.rs#L127) code I don't use `load_defaults`.